client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/privileges/Role.java

```diff
@@ -156,14 +160,13 @@ public boolean equals(Object o) {
             && indicesPrivileges.equals(that.indicesPrivileges)
             && applicationResourcePrivileges.equals(that.applicationResourcePrivileges)
             && runAsPrivilege.equals(that.runAsPrivilege)
-            && metadata.equals(that.metadata)
-            && transientMetadata.equals(that.transientMetadata);
+            && metadata.equals(that.metadata);
     }
 
     @Override
     public int hashCode() {
         return Objects.hash(name, clusterPrivileges, globalApplicationPrivileges, indicesPrivileges, applicationResourcePrivileges,
-            runAsPrivilege, metadata, transientMetadata);
+            runAsPrivilege, metadata);
```

Comment:
transient metadata is a read only field when retrieving the Role and cannot be set on a put Role.
It should not matter in an equals, because a user should be able to create a role that is equal to another one, except for the transient part.

Reply:
Is a (retrieved) enabled role equal (for a client) to a disabled role ?

Reply:
> In principle yes because the enabled status is included in the transientMetadata which is not counted by equals.

Yes, this is what we have now. My (poorly stated) stated question was whether we think it should be this way or not. 

>  In practice I don't know how the client could get hold of a disabled role (probably before disabling it).

A role that has DLS/FLS for a user that was on platinum but is now on Gold maybe. 

> Do you have some scenario in mind when you think transientMetadata would be something the client should touch?

No not specifically. I was just wondering if the client should be able to distinguish between the same role being enabled or disabled by comparing the role objects

Reply:
This is similar to the `enabled` flag on the user. I think the `Role` that is included in a `PutRoleRequest` should contain only fields that the client can change. `transientMetadata` is a field that is read only, and it is a compromise because the `Role` is also part of the `GetRolesResponse` . We can completely pull out `transientMetadata` from the `Role` and present it other way in the `Get Roles API`. `GetRolesResponse` would be having some sort of transientMetadata field. Do you see this preferable?

Reply:
I think it's clearer.. I'm now circling back to our discussion regarding `Role` when I started with the GetRoles API and it's much clearer what you were saying back then. I probably should have done it in the get roles effort but it wasn't obvious at that point, sorry :heart:

Reply:
I think it's clearer too. I will make the change in this PR, instead of opening a new one, because `Role` was "synthetically" created and was expected to change as APIs suit it. But we need to get this PR in before 6.6 so as not to be breaking because it will change `GetRolesResponse`.
Sounds like a plan?

Reply:
@bizybot I prefer to pass the Role to the `PutRoleRequest` and not take fields apart to call builder methods.